src/Microsoft.Performance.SDK.Runtime/VersionChecker.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using NuGet.Versioning;
+
+namespace Microsoft.Performance.SDK.Runtime
+{
+    /// <summary>
+    ///     This class is used to check compatibility of different
+    ///     versions of the SDK. This class is only meant to be used
+    ///     from the hosting process. The hosting process's version of
+    ///     the SDK is used as the reference for all compatibility checks.
+    /// </summary>
+    public class VersionChecker
+    {
+        private static readonly SemanticVersion SdkVersion;
+
+        private static readonly SemanticVersion LowestSupportedSdkVersion;
+
+        /// <summary>
+        ///     Initializes the static members of the <see cref="VersionChecker"/>
+        ///     class.
+        /// </summary>
+        static VersionChecker()
+        {
+            SdkVersion = SdkAssembly.Assembly.GetSemanticVersion();
+
+            //
+            // This is the lowest version of the SDK that is
+            // compatible with SdkVersion. If any breaking changes occur
+            // before the 1.0 release, then this value will be updated.
+            //
+
+            LowestSupportedSdkVersion = new SemanticVersion(0, 109, 0);
+        }
+
+        protected VersionChecker()
+        {
+        }
+
+        /// <summary>
+        ///     Gets the Version of the SDK that is being hosted.
+        /// </summary>
+        public virtual SemanticVersion Sdk => SdkVersion;
+
+        /// <summary>
+        ///     Gets the minimum supported Version of the SDK.
+        /// </summary>
+        public virtual SemanticVersion LowestSupportedSdk => LowestSupportedSdkVersion;
+
+        /// <summary>
+        ///     Creates a new <see cref="VersionChecker"/>.
+        /// </summary>
+        /// <returns>
+        ///     A new instance of the <see cref="VersionChecker"/> class.
+        /// </returns>
+        public static VersionChecker Create()
+        {
+            return new VersionChecker();
+        }
+
+        /// <summary>
+        ///     Determines whether the given Version of the SDK is
+        ///     supported by the host.
+        /// </summary>
+        /// <param name="candidateVersion">
+        ///     The version of the SDK being used by the Plugin.
+        /// </param>
+        /// <returns>
+        ///     <c>true</c> if the Plugin's version of the SDK is
+        ///     compatible with this version of the SDK; <c>false</c>
+        ///     otherwise.
+        /// </returns>
+        public virtual bool IsVersionSupported(SemanticVersion candidateVersion)
```

Comment:
Is this and the below method virtual on purpose? Can't imagine a deriving class wanting to change that logic

Reply:
I think it was done for testing.

---
In reply to: [574916892](https://github.com/microsoft/microsoft-performance-toolkit-sdk/pull/39#discussion_r574916892) [](ancestors = 574916892)